File: src/main/java/net/moddingplayground/direbats/item/DirebatFangArrowItem.java
package net.moddingplayground.direbats.item;

import net.minecraft.entity.LivingEntity;
import net.minecraft.entity.projectile.PersistentProjectileEntity;
import net.minecraft.item.ArrowItem;
import net.minecraft.item.Item;
import net.minecraft.item.ItemStack;
import net.minecraft.world.World;
import net.moddingplayground.direbats.entity.DirebatFangArrowEntity;

public class DirebatFangArrowItem extends ArrowItem {
	public DirebatFangArrowItem(Item.Settings settings) {
        super(settings);
    }

    @Override
    public PersistentProjectileEntity createArrow(World world, ItemStack stack, LivingEntity shooter) {
        return new DirebatFangArrowEntity(world, shooter);
    }
}

File: src/main/kotlin/dev/andante/direbats/item/DirebatFangArrowItem.kt
package dev.andante.direbats.item

import dev.andante.direbats.entity.DirebatFangArrowEntity
import net.minecraft.entity.LivingEntity
import net.minecraft.entity.projectile.PersistentProjectileEntity
import net.minecraft.item.ArrowItem
import net.minecraft.item.ItemStack
import net.minecraft.world.World

/**
 * Represents a Direbat Fang Arrow item.
 * @see [DirebatsItems.DIREBAT_FANG_ARROW]
 */
class DirebatFangArrowItem(settings: Settings) : ArrowItem(settings) {
    override fun createArrow(world: World, stack: ItemStack, shooter: LivingEntity): PersistentProjectileEntity {
        return DirebatFangArrowEntity(world, shooter)
    }
}
